crates/oxc_linter/src/rules/unicorn/no_empty_file.rs

```diff
@@ -0,0 +1,145 @@
+use oxc_ast::{ast::Statement, AstKind};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{context::LintContext, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-unicorn(no-empty-file): Empty files are not allowed.")]
+#[diagnostic(severity(warning), help("Delete this file or add some code to it."))]
+struct NoEmptyFileDiagnostic(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoEmptyFile;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    ///
+    /// Disallows any files only containing the following:
+    ///  - Whitespace
+    ///  - Comments
+    ///  - Directives
+    ///  - Empty statements
+    ///  - Empty blocks
+    ///  - Hashbang
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Meaningless files clutter a codebase.
+    ///
+    NoEmptyFile,
+    restriction,
+);
+
+impl Rule for NoEmptyFile {
+    fn run_once(&self, ctx: &LintContext) {
+        println!("~~~~");
+        let Some(root) = ctx.nodes().iter().next() else { return };
```

Comment:
This is smart!

Reply:
thanks! we could probably unwrap those two lines. but probably better to be safe 🙂